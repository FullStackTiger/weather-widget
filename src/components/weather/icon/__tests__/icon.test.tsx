import { render } from "@testing-library/react";
import { Icon } from "..";

describe("WeatherIcon", () => {
  it("should return Thunderstorm Icon", () => {
    const { container } = render(<Icon id={200} />);
    expect(container).toMatchSnapshot();
  });

  it("should return ShowerRain Icon", () => {
    const { container } = render(<Icon id={300} />);
    expect(container).toMatchSnapshot();
  });

  it("should return Rain Icon", () => {
    const { container } = render(<Icon id={500} />);
    expect(container).toMatchSnapshot();
  });

  it("should return Snow Icon", () => {
    const { container } = render(<Icon id={600} />);
    expect(container).toMatchSnapshot();
  });

  it("should return Mist Icon", () => {
    const { container } = render(<Icon id={701} />);
    expect(container).toMatchSnapshot();
  });

  it("should return ClearSky Icon", () => {
    const { container } = render(<Icon id={800} />);
    expect(container).toMatchSnapshot();
  });

  it("should return FewClouds Icon", () => {
    const { container } = render(<Icon id={801} />);
    expect(container).toMatchSnapshot();
  });

  it("should return ScatteredClouds Icon", () => {
    const { container } = render(<Icon id={802} />);
    expect(container).toMatchSnapshot();
  });

  it("should return null", () => {
    const { container } = render(<Icon id={102} />);
    expect(container).toMatchSnapshot();
  });
});
