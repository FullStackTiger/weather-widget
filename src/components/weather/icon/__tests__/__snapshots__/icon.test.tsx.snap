// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherIcon should return ClearSky Icon 1`] = `
<div>
  <svg
    fill="none"
    height="380"
    viewBox="0 0 380 380"
    width="380"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M190.098 290.098C134.958 290.098 90.0977 245.238 90.0977 190.098C90.0977 134.958 134.958 90.0977 190.098 90.0977C245.238 90.0977 290.098 134.958 290.098 190.098C290.098 245.238 245.238 290.098 190.098 290.098ZM190.098 110.098C145.985 110.098 110.098 145.985 110.098 190.098C110.098 234.21 145.985 270.098 190.098 270.098C234.21 270.098 270.098 234.21 270.098 190.098C270.098 145.985 234.21 110.098 190.098 110.098Z"
      fill="white"
    />
    <path
      d="M190 74C184.477 74 180 69.5229 180 64V10C180 4.47705 184.477 0 190 0C195.523 0 200 4.47705 200 10V64C200 69.5229 195.523 74 190 74Z"
      fill="white"
    />
    <path
      d="M190 380C184.477 380 180 375.523 180 370V316C180 310.477 184.477 306 190 306C195.523 306 200 310.477 200 316V370C200 375.523 195.523 380 190 380Z"
      fill="white"
    />
    <path
      d="M64 200H10C4.47705 200 0 195.523 0 190C0 184.477 4.47705 180 10 180H64C69.5229 180 74 184.477 74 190C74 195.523 69.5229 200 64 200Z"
      fill="white"
    />
    <path
      d="M370 200H316C310.477 200 306 195.523 306 190C306 184.477 310.477 180 316 180H370C375.523 180 380 184.477 380 190C380 195.523 375.523 200 370 200Z"
      fill="white"
    />
    <path
      d="M278.542 111.458C275.983 111.458 273.423 110.482 271.471 108.529C267.566 104.624 267.566 98.292 271.471 94.3867L310.208 55.6494C314.113 51.7441 320.445 51.7441 324.35 55.6494C328.255 59.5547 328.255 65.8867 324.35 69.792L285.613 108.529C283.661 110.482 281.101 111.458 278.542 111.458Z"
      fill="white"
    />
    <path
      d="M62.7206 327.279C60.1615 327.279 57.6019 326.303 55.6493 324.351C51.7445 320.445 51.7445 314.113 55.6493 310.208L94.3866 271.471C98.2919 267.565 104.624 267.565 108.529 271.471C112.434 275.376 112.434 281.708 108.529 285.613L69.7919 324.351C67.8392 326.303 65.2797 327.279 62.7206 327.279Z"
      fill="white"
    />
    <path
      d="M101.458 111.458C98.8988 111.458 96.3392 110.482 94.3866 108.529L55.6493 69.792C51.7445 65.8867 51.7445 59.5547 55.6493 55.6494C59.5546 51.7441 65.8866 51.7441 69.7919 55.6494L108.529 94.3867C112.434 98.292 112.434 104.624 108.529 108.529C106.577 110.482 104.017 111.458 101.458 111.458Z"
      fill="white"
    />
    <path
      d="M317.279 327.279C314.72 327.279 312.161 326.303 310.208 324.351L271.471 285.613C267.566 281.708 267.566 275.376 271.471 271.471C275.376 267.565 281.708 267.565 285.613 271.471L324.35 310.208C328.255 314.113 328.255 320.445 324.35 324.351C322.398 326.303 319.838 327.279 317.279 327.279Z"
      fill="white"
    />
  </svg>
</div>
`;

exports[`WeatherIcon should return FewClouds Icon 1`] = `
<div>
  <svg
    fill="none"
    height="420"
    viewBox="0 0 460 420"
    width="460"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M326.798 270C323.89 270 321.006 268.738 319.029 266.304C315.548 262.017 316.201 255.719 320.488 252.237C339.306 236.957 350.098 214.307 350.098 190.098C350.098 145.985 314.21 110.098 270.098 110.098C229.294 110.098 195.117 140.626 190.596 181.109C189.983 186.598 185.039 190.548 179.548 189.938C174.06 189.325 170.107 184.378 170.72 178.89C176.372 128.27 219.095 90.0977 270.098 90.0977C325.238 90.0977 370.098 134.958 370.098 190.098C370.098 220.359 356.611 248.667 333.096 267.763C331.242 269.268 329.013 270 326.798 270Z"
      fill="white"
    />
    <path
      d="M270 74C264.478 74 260 69.5229 260 64V10C260 4.47705 264.478 0 270 0C275.522 0 280 4.47705 280 10V64C280 69.5229 275.522 74 270 74Z"
      fill="white"
    />
    <path
      d="M144 200H90C84.477 200 80 195.523 80 190C80 184.477 84.477 180 90 180H144C149.523 180 154 184.477 154 190C154 195.523 149.523 200 144 200Z"
      fill="white"
    />
    <path
      d="M450 200H396C390.478 200 386 195.523 386 190C386 184.477 390.478 180 396 180H450C455.522 180 460 184.477 460 190C460 195.523 455.522 200 450 200Z"
      fill="white"
    />
    <path
      d="M358.542 111.458C355.982 111.458 353.424 110.482 351.471 108.529C347.565 104.624 347.565 98.292 351.471 94.3867L390.208 55.6494C394.114 51.7441 400.444 51.7441 404.351 55.6494C408.256 59.5547 408.256 65.8867 404.351 69.792L365.613 108.529C363.66 110.482 361.102 111.458 358.542 111.458Z"
      fill="white"
    />
    <path
      d="M181.458 111.458C178.899 111.458 176.339 110.482 174.387 108.529L135.649 69.792C131.745 65.8867 131.745 59.5547 135.649 55.6494C139.555 51.7441 145.887 51.7441 149.792 55.6494L188.529 94.3867C192.434 98.292 192.434 104.624 188.529 108.529C186.577 110.482 184.017 111.458 181.458 111.458Z"
      fill="white"
    />
    <path
      d="M397.279 327.279C394.72 327.279 392.161 326.303 390.208 324.351L351.471 285.613C347.565 281.708 347.565 275.376 351.471 271.471C355.377 267.565 361.707 267.565 365.613 271.471L404.351 310.208C408.256 314.113 408.256 320.445 404.351 324.351C402.397 326.303 399.839 327.279 397.279 327.279Z"
      fill="white"
    />
    <path
      d="M69.9668 420C31.4951 419.747 0.25 388.342 0.25 349.994C0.25 316.669 23.2368 288.814 54.5288 281.731C54.02 278.932 53.7646 276.075 53.7646 273.172C53.7646 246.919 75.1226 225.562 101.375 225.562C110.095 225.562 118.453 227.889 125.745 232.25C145.726 205.827 176.557 190.352 210.206 190.352C262.794 190.352 307.121 229.09 314.986 280.136C351.385 282.588 380.25 312.984 380.25 349.994C380.25 388.216 349.154 419.618 310.932 419.995L69.9668 420ZM101.375 245.562C86.1509 245.562 73.7646 257.947 73.7646 273.172C73.7646 277.515 74.7505 281.679 76.6948 285.546C78.2217 288.583 78.1011 292.188 76.3745 295.117C74.6479 298.045 71.5518 299.896 68.1548 300.03C41.2925 301.09 20.25 323.037 20.25 349.994C20.25 377.386 42.5825 399.819 70.0327 400L310.833 399.995C338.037 399.727 360.25 377.296 360.25 349.994C360.25 322.414 337.812 299.977 310.233 299.977C309.206 299.977 308.089 300.021 306.719 300.116C304.072 300.297 301.444 299.42 299.441 297.667C297.438 295.915 296.216 293.437 296.046 290.78C293.152 245.68 255.447 210.352 210.206 210.352C179.866 210.352 152.347 225.861 136.592 251.838C134.997 254.468 132.28 256.217 129.226 256.582C126.172 256.942 123.12 255.885 120.951 253.704C115.729 248.453 108.777 245.562 101.375 245.562Z"
      fill="white"
    />
  </svg>
</div>
`;

exports[`WeatherIcon should return Mist Icon 1`] = `
<div>
  <svg
    fill="none"
    height="248"
    viewBox="0 0 351 248"
    width="351"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M290 136H36.918C17.2461 136 0.625977 119.061 0.625977 99.0107C0.625977 78.6147 16.8799 62.021 36.8584 62.021C56.8213 62.021 73.0615 78.5059 73.0615 98.7681C73.0615 104.291 68.584 108.768 63.0615 108.768C57.5391 108.768 53.0615 104.291 53.0615 98.7681C53.0615 89.6904 45.6416 82.021 36.8584 82.021C28.0596 82.021 20.626 89.8013 20.626 99.0107C20.626 108.22 28.0869 116 36.918 116H290C295.522 116 300 120.477 300 126C300 131.523 295.522 136 290 136Z"
      fill="white"
    />
    <path
      d="M341 93H136.138C111.012 93 90.5703 72.3569 90.5703 46.9829C90.5703 21.6089 110.919 0.965332 135.931 0.965332C160.886 0.965332 181.188 21.5259 181.188 46.7983C181.188 52.3213 176.71 56.7983 171.188 56.7983C165.665 56.7983 161.188 52.3213 161.188 46.7983C161.188 32.5542 149.857 20.9653 135.931 20.9653C121.947 20.9653 110.57 32.6367 110.57 46.9829C110.57 61.0854 122.278 73 136.138 73H341C346.522 73 351 77.4771 351 83C351 88.5229 346.522 93 341 93Z"
      fill="white"
    />
    <path
      d="M69.3643 247.823C57.2256 247.823 45.8428 243.217 37.3115 234.853C28.6963 226.406 23.9512 215.062 23.9512 202.912C23.9512 178.147 44.3691 158 69.4658 158H319C324.522 158 329 162.478 329 168C329 173.522 324.522 178 319 178H69.4658C55.1582 178 43.9512 188.942 43.9512 202.912C43.9512 216.881 55.1143 227.823 69.3643 227.823C83.5859 227.823 94.7266 216.8 94.7266 202.728C94.7266 197.205 99.2041 192.728 104.727 192.728C110.249 192.728 114.727 197.205 114.727 202.728C114.727 227.594 94.3769 247.823 69.3643 247.823Z"
      fill="white"
    />
  </svg>
</div>
`;

exports[`WeatherIcon should return Rain Icon 1`] = `
<div>
  <svg
    fill="none"
    height="337"
    viewBox="0 0 380 337"
    width="380"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M310.58 230.091L299.996 230.087L69.7168 230C31.2749 230 0 198.639 0 160.089C0 126.813 22.9863 98.9956 54.2788 91.9214C53.7705 89.1226 53.5146 86.2661 53.5146 83.3638C53.5146 57.1108 74.8726 35.7524 101.125 35.7524C109.846 35.7524 118.203 38.0786 125.495 42.4395C145.477 16.0147 176.309 0.538086 209.956 0.538086C262.541 0.538086 306.865 39.2666 314.734 90.3042C351.136 92.7524 380 123.138 380 160.138C380 198.334 348.904 229.715 310.682 230.09C310.648 230.091 310.614 230.091 310.58 230.091ZM300.003 210.087L310.535 210.091C337.815 209.796 360 187.397 360 160.138C360 132.572 337.562 110.145 309.983 110.145C308.805 110.145 307.545 110.222 306.468 110.296C303.816 110.483 301.194 109.6 299.191 107.847C297.188 106.095 295.966 103.616 295.795 100.96C292.903 55.8638 255.197 20.5381 209.956 20.5381C179.617 20.5381 152.098 36.0483 136.342 62.0273C134.747 64.6572 132.03 66.4067 128.976 66.7715C125.921 67.1313 122.87 66.0747 120.701 63.8936C115.48 58.6436 108.528 55.7524 101.125 55.7524C85.9009 55.7524 73.5146 68.1387 73.5146 83.3638C73.5146 87.7075 74.5005 91.8706 76.4448 95.7378C77.9717 98.7749 77.8511 102.38 76.1245 105.309C74.3979 108.237 71.3018 110.087 67.9048 110.222C41.0425 111.282 20 133.187 20 160.089C20 187.61 42.3027 210 69.7168 210H80L300.003 210.087Z"
      fill="white"
    />
    <path
      class="rain-path"
      d="M154.958 336.939C152.397 336.939 149.836 335.962 147.883 334.007C143.98 330.1 143.983 323.768 147.891 319.865L172.933 294.849C176.84 290.945 183.172 290.948 187.075 294.855C190.978 298.763 190.975 305.095 187.067 308.997L162.025 334.014C160.073 335.964 157.515 336.939 154.958 336.939Z"
      fill="white"
      id="rain-4"
    />
    <path
      class="rain-path"
      d="M67.958 279.939C65.397 279.939 62.8364 278.962 60.8833 277.007C56.98 273.1 56.9834 266.769 60.8906 262.865L76.6826 247.089C80.5898 243.186 86.9214 243.188 90.8247 247.096C94.728 251.003 94.7246 257.334 90.8174 261.237L75.0254 277.014C73.0732 278.964 70.5156 279.939 67.958 279.939Z"
      fill="white"
      id="rain-1"
    />
    <path
      class="rain-path"
      d="M206.958 279.939C204.397 279.939 201.836 278.962 199.883 277.007C195.98 273.1 195.983 266.769 199.891 262.865L215.683 247.089C219.59 243.186 225.921 243.188 229.825 247.096C233.728 251.003 233.725 257.334 229.817 261.237L214.025 277.014C212.073 278.964 209.516 279.939 206.958 279.939Z"
      fill="white"
      id="rain-2"
    />
    <path
      class="rain-path"
      d="M251.958 305.103C249.396 305.103 246.835 304.125 244.882 302.169C240.979 298.262 240.984 291.93 244.892 288.027L285.884 247.088C289.792 243.185 296.123 243.188 300.026 247.097C303.929 251.004 303.924 257.336 300.017 261.238L259.024 302.178C257.072 304.128 254.515 305.103 251.958 305.103Z"
      fill="white"
      id="rain-3"
    />
    <path
      class="rain-path"
      d="M100.958 316.103C98.397 316.103 95.8364 315.125 93.8833 313.17C89.98 309.263 89.9834 302.932 93.8906 299.028L145.883 247.089C149.79 243.186 156.122 243.188 160.025 247.096C163.928 251.003 163.925 257.334 160.018 261.237L108.025 313.177C106.073 315.127 103.515 316.103 100.958 316.103Z"
      fill="white"
      id="rain-5"
    />
  </svg>
</div>
`;

exports[`WeatherIcon should return ScatteredClouds Icon 1`] = `
<div>
  <svg
    fill="none"
    height="231"
    viewBox="0 0 380 231"
    width="380"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M69.7168 230.192C31.2451 229.939 0 198.534 0 160.186C0 126.862 22.9868 99.0063 54.2788 91.9233C53.77 89.124 53.5146 86.2676 53.5146 83.3643C53.5146 57.1118 74.8726 35.7539 101.125 35.7539C109.845 35.7539 118.203 38.0811 125.495 42.4419C145.476 16.019 176.307 0.544434 209.956 0.544434C262.544 0.544434 306.871 39.2827 314.736 90.3286C351.134 92.7808 380 123.176 380 160.186C380 198.408 348.904 229.811 310.682 230.187L69.7168 230.192ZM101.125 55.7539C85.9009 55.7539 73.5146 68.1396 73.5146 83.3643C73.5146 87.7075 74.5005 91.8711 76.4448 95.7383C77.9717 98.7754 77.8511 102.38 76.1245 105.309C74.3979 108.237 71.3018 110.088 67.9048 110.222C41.0425 111.283 20 133.229 20 160.186C20 187.579 42.3325 210.012 69.7827 210.192L310.583 210.188C337.788 209.919 360 187.488 360 160.186C360 132.606 337.562 110.169 309.983 110.169C308.956 110.169 307.838 110.213 306.468 110.309C303.814 110.494 301.194 109.611 299.191 107.859C297.188 106.107 295.966 103.628 295.795 100.973C292.902 55.8726 255.197 20.5444 209.956 20.5444C179.616 20.5444 152.097 36.0532 136.342 62.0303C134.747 64.6602 132.03 66.4097 128.976 66.7744C125.923 67.1348 122.87 66.0776 120.701 63.8965C115.479 58.6455 108.527 55.7539 101.125 55.7539Z"
      fill="white"
    />
  </svg>
</div>
`;

exports[`WeatherIcon should return ShowerRain Icon 1`] = `
<div>
  <svg
    fill="none"
    height="473"
    viewBox="0 0 416 473"
    width="416"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M301.532 233.5C299.017 233.5 296.506 232.508 294.638 230.539C291.027 226.732 291.186 220.719 294.992 217.108C309.482 203.363 317.463 184.798 317.463 164.833C317.463 125.111 285.147 92.7954 245.425 92.7954C210.536 92.7954 180.724 117.668 174.536 151.938C173.604 157.1 168.659 160.529 163.5 159.599C158.336 158.667 154.906 153.725 155.839 148.562C163.66 105.239 201.337 73.7954 245.425 73.7954C295.623 73.7954 336.463 114.635 336.463 164.833C336.463 190.063 326.379 213.523 308.068 230.893C306.232 232.636 303.879 233.5 301.532 233.5Z"
      fill="white"
    />
    <path
      d="M217.268 65.8277C212.898 65.8277 208.965 62.795 207.996 58.3506L197.977 12.4307C196.858 7.30471 200.107 2.2427 205.233 1.12405C210.369 0.00442147 215.423 3.2554 216.54 8.38089L226.558 54.3008C227.678 59.4268 224.429 64.4888 219.303 65.6075C218.62 65.7564 217.938 65.8277 217.268 65.8277Z"
      fill="white"
    />
    <path
      d="M356.9 154.605C352.53 154.605 348.597 151.572 347.628 147.128C346.509 142.002 349.758 136.94 354.884 135.821L403.734 125.164C408.871 124.044 413.924 127.295 415.041 132.42C416.16 137.546 412.911 142.608 407.785 143.727L358.934 154.385C358.252 154.534 357.57 154.605 356.9 154.605Z"
      fill="white"
    />
    <path
      d="M305.773 81.8574C304.016 81.8574 302.238 81.3711 300.65 80.3515C296.234 77.5176 294.952 71.6406 297.786 67.2251L324.595 25.456C327.429 21.0405 333.305 19.7588 337.722 22.5923C342.137 25.4263 343.419 31.3032 340.585 35.7187L313.776 77.4878C311.962 80.3149 308.9 81.8574 305.773 81.8574Z"
      fill="white"
    />
    <path
      d="M149.025 116.055C147.269 116.055 145.491 115.568 143.903 114.549L102.134 87.7407C97.7178 84.9067 96.4356 79.0298 99.2696 74.6142C102.104 70.1987 107.981 68.917 112.396 71.7505L154.165 98.5586C158.581 101.393 159.863 107.27 157.029 111.685C155.215 114.512 152.152 116.055 149.025 116.055Z"
      fill="white"
    />
    <path
      d="M380.762 265.626C379.005 265.626 377.228 265.14 375.639 264.121L333.87 237.313C329.455 234.479 328.173 228.602 331.007 224.186C333.841 219.771 339.717 218.487 344.134 221.322L385.902 248.13C390.317 250.964 391.6 256.841 388.766 261.257C386.951 264.084 383.889 265.626 380.762 265.626Z"
      fill="white"
    />
    <path
      d="M64.1104 373.561C29.0117 373.33 0.507812 344.68 0.507812 309.696C0.507812 279.44 21.2783 254.126 49.6172 247.516C49.2002 245.087 48.9912 242.611 48.9912 240.097C48.9912 216.07 68.5381 196.523 92.5654 196.523C100.411 196.523 107.938 198.582 114.53 202.444C132.724 178.586 160.673 164.624 191.164 164.624C238.908 164.624 279.171 199.703 286.474 245.993C319.515 248.399 345.661 276.051 345.661 309.696C345.661 344.564 317.293 373.212 282.425 373.556L64.1104 373.561ZM92.5654 215.523C79.0156 215.523 67.9912 226.547 67.9912 240.097C67.9912 243.963 68.8682 247.668 70.5986 251.109C72.0498 253.995 71.9355 257.419 70.2949 260.202C68.6553 262.984 65.7139 264.742 62.4863 264.87C38.3857 265.821 19.5078 285.511 19.5078 309.696C19.5078 334.272 39.5449 354.398 64.1729 354.561L282.331 354.556C306.732 354.315 326.661 334.191 326.661 309.696C326.661 284.953 306.53 264.822 281.787 264.822C280.861 264.822 279.859 264.862 278.632 264.947C276.102 265.126 273.621 264.284 271.72 262.62C269.816 260.955 268.656 258.601 268.494 256.078C265.888 215.449 231.921 183.624 191.164 183.624C163.833 183.624 139.041 197.595 124.848 220.998C123.332 223.496 120.752 225.158 117.851 225.504C114.947 225.849 112.049 224.841 109.989 222.771C105.341 218.097 99.1523 215.523 92.5654 215.523Z"
      fill="white"
    />
    <path
      d="M147.452 472.822C145.02 472.822 142.587 471.893 140.731 470.036C137.023 466.324 137.026 460.309 140.738 456.601L163.426 433.937C167.138 430.229 173.152 430.231 176.86 433.943C180.568 437.655 180.565 443.67 176.854 447.378L154.166 470.042C152.312 471.896 149.882 472.822 147.452 472.822Z"
      fill="white"
    />
    <path
      d="M68.6318 421.181C66.1992 421.181 63.7666 420.252 61.9111 418.395C58.2031 414.683 58.2061 408.668 61.918 404.96L76.2256 390.667C79.9365 386.96 85.9512 386.962 89.6602 390.674C93.3682 394.386 93.3652 400.401 89.6533 404.109L75.3457 418.402C73.4912 420.255 71.0615 421.181 68.6318 421.181Z"
      fill="white"
    />
    <path
      d="M194.563 421.181C192.131 421.181 189.698 420.253 187.843 418.396C184.135 414.684 184.138 408.668 187.85 404.96L202.156 390.667C205.868 386.96 211.883 386.962 215.591 390.674C219.299 394.386 219.296 400.401 215.584 404.109L201.277 418.402C199.423 420.254 196.993 421.181 194.563 421.181Z"
      fill="white"
    />
    <path
      d="M235.333 443.979C232.899 443.979 230.467 443.049 228.611 441.192C224.903 437.479 224.907 431.464 228.62 427.757L265.758 390.667C269.47 386.958 275.484 386.963 279.192 390.675C282.9 394.387 282.896 400.402 279.183 404.11L242.046 441.2C240.191 443.053 237.763 443.979 235.333 443.979Z"
      fill="white"
    />
    <path
      d="M98.5293 453.944C96.0967 453.944 93.6641 453.016 91.8086 451.159C88.1006 447.447 88.1035 441.432 91.8154 437.724L138.919 390.667C142.631 386.959 148.646 386.963 152.354 390.674C156.062 394.386 156.059 400.401 152.347 404.109L105.243 451.165C103.389 453.018 100.959 453.944 98.5293 453.944Z"
      fill="white"
    />
  </svg>
</div>
`;

exports[`WeatherIcon should return Snow Icon 1`] = `
<div>
  <svg
    fill="none"
    height="384"
    viewBox="0 0 380 384"
    width="380"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M310.579 230.091L299.995 230.087L69.7168 230C31.2749 230 0 198.639 0 160.089C0 126.813 22.9863 98.9956 54.2788 91.9214C53.7705 89.1226 53.5146 86.2661 53.5146 83.3638C53.5146 57.1108 74.8726 35.7524 101.125 35.7524C109.846 35.7524 118.203 38.0786 125.495 42.4395C145.477 16.0147 176.309 0.538086 209.956 0.538086C262.54 0.538086 306.865 39.2666 314.734 90.3042C351.137 92.7524 380 123.138 380 160.138C380 198.334 348.903 229.715 310.682 230.09C310.647 230.091 310.613 230.091 310.579 230.091ZM300.003 210.087L310.534 210.091C337.814 209.796 360 187.397 360 160.138C360 132.572 337.562 110.145 309.983 110.145C308.805 110.145 307.545 110.222 306.469 110.296C303.826 110.479 301.195 109.6 299.191 107.847C297.188 106.095 295.966 103.616 295.796 100.96C292.902 55.8638 255.197 20.5381 209.956 20.5381C179.617 20.5381 152.098 36.0483 136.342 62.0273C134.747 64.6572 132.03 66.4067 128.976 66.7715C125.924 67.1338 122.87 66.0747 120.701 63.8936C115.48 58.6436 108.528 55.7524 101.125 55.7524C85.9009 55.7524 73.5146 68.1387 73.5146 83.3638C73.5146 87.7075 74.5005 91.8706 76.4448 95.7378C77.9717 98.7749 77.8511 102.38 76.1245 105.309C74.3979 108.237 71.3018 110.087 67.9048 110.222C41.0425 111.282 20 133.187 20 160.089C20 187.61 42.3027 210 69.7168 210H80L300.003 210.087Z"
      fill="white"
    />
    <path
      d="M94.0024 365.054C91.4414 365.054 88.8809 364.076 86.9277 362.121C83.0244 358.214 83.0278 351.883 86.9351 347.979L148.838 286.139C152.746 282.236 159.077 282.238 162.98 286.146C166.884 290.053 166.88 296.384 162.973 300.287L101.07 362.128C99.1177 364.078 96.5601 365.054 94.0024 365.054Z"
      fill="white"
    />
    <path
      d="M155.875 365.085C153.313 365.085 150.753 364.107 148.8 362.152L86.959 300.249C83.0557 296.342 83.0591 290.011 86.9663 286.107C90.874 282.205 97.2056 282.207 101.108 286.114L162.949 348.018C166.853 351.925 166.849 358.256 162.942 362.159C160.99 364.109 158.432 365.085 155.875 365.085Z"
      fill="white"
    />
    <path
      d="M174.954 334C174.952 334 174.951 334 174.949 334L74.9492 334.029C69.4263 334.026 64.9512 329.532 64.9541 324.01C64.957 318.488 69.4336 314 74.9541 314C74.9556 314 74.9575 314 74.959 314L174.959 314.104C180.482 314.107 184.957 318.534 184.954 324.057C184.951 329.577 180.474 334 174.954 334Z"
      fill="white"
    />
    <path
      d="M124.929 384C124.927 384 124.926 384 124.924 384C119.401 384 114.926 379.584 114.929 374.062L114.979 274.028C114.982 268.507 119.459 264 124.979 264C124.981 264 124.982 264 124.984 264C130.507 264 134.982 268.549 134.979 274.071L134.929 374.038C134.926 379.56 130.449 384 124.929 384Z"
      fill="white"
    />
    <path
      d="M232.711 320.734C229.266 320.734 225.914 318.953 224.058 315.761C221.28 310.987 222.898 304.866 227.673 302.089L288.179 266.89C292.953 264.108 299.074 265.73 301.851 270.505C304.628 275.278 303.01 281.399 298.235 284.177L237.729 319.376C236.147 320.297 234.418 320.734 232.711 320.734Z"
      fill="white"
    />
    <path
      d="M293.707 319.83C292.096 319.83 290.462 319.439 288.945 318.617L227.422 285.227C222.569 282.592 220.769 276.521 223.403 271.667C226.037 266.813 232.107 265.015 236.963 267.648L298.486 301.039C303.34 303.674 305.14 309.744 302.505 314.599C300.694 317.936 297.256 319.83 293.707 319.83Z"
      fill="white"
    />
    <path
      d="M262.937 338C262.935 338 262.934 338 262.932 338C257.408 338 252.934 333.29 252.937 327.768L252.972 258.97C252.975 253.448 257.451 249 262.972 249C262.974 249 262.975 249 262.977 249C268.5 249 272.975 253.343 272.972 258.866L272.937 327.847C272.934 333.368 268.457 338 262.937 338Z"
      fill="white"
    />
  </svg>
</div>
`;

exports[`WeatherIcon should return Thunderstorm Icon 1`] = `
<div>
  <svg
    fill="none"
    height="337"
    viewBox="0 0 380 337"
    width="380"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M310.58 230.091L299.996 230.087C294.473 230.085 289.998 225.606 290 220.083C290.001 214.562 294.479 210.087 300 210.087H300.003L310.535 210.091C337.815 209.796 360 187.397 360 160.138C360 132.572 337.562 110.145 309.983 110.145C308.805 110.145 307.545 110.222 306.468 110.296C303.814 110.483 301.194 109.6 299.191 107.847C297.188 106.095 295.966 103.616 295.795 100.96C292.903 55.8638 255.197 20.5381 209.956 20.5381C179.617 20.5381 152.098 36.0483 136.342 62.0273C134.747 64.6572 132.03 66.4067 128.976 66.7715C125.923 67.1313 122.87 66.0747 120.701 63.8936C115.48 58.6436 108.528 55.7524 101.125 55.7524C85.9009 55.7524 73.5146 68.1387 73.5146 83.3638C73.5146 87.7075 74.5005 91.8706 76.4449 95.7378C77.9717 98.7749 77.8511 102.38 76.1245 105.309C74.398 108.237 71.3018 110.087 67.9048 110.222C41.0425 111.282 20 133.187 20 160.089C20 187.61 42.3027 210 69.7168 210H80C85.5229 210 90 214.478 90 220C90 225.522 85.5229 230 80 230H69.7168C31.2749 230 0 198.639 0 160.089C0 126.813 22.9863 98.9956 54.2788 91.9214C53.7705 89.1226 53.5146 86.2661 53.5146 83.3638C53.5146 57.1108 74.8726 35.7524 101.125 35.7524C109.846 35.7524 118.203 38.0786 125.495 42.4395C145.477 16.0147 176.309 0.538086 209.956 0.538086C262.541 0.538086 306.865 39.2666 314.734 90.3042C351.136 92.7524 380 123.138 380 160.138C380 198.334 348.904 229.715 310.682 230.09C310.648 230.091 310.614 230.091 310.58 230.091Z"
      fill="white"
    />
    <path
      d="M96.0332 324.085C93.4761 324.085 90.9194 323.11 88.9673 321.161C85.0591 317.259 85.0547 310.927 88.957 307.019L131.915 264H106.033C101.941 264 98.2627 261.508 96.7456 257.707C95.229 253.907 96.1802 249.566 99.147 246.749L162.147 186.92C166.152 183.117 172.481 183.278 176.284 187.285C180.087 191.289 179.924 197.619 175.919 201.422L131.085 244H156.033C160.077 244 163.722 246.435 165.271 250.17C166.819 253.905 165.966 258.205 163.109 261.066L103.109 321.151C101.156 323.106 98.5947 324.085 96.0332 324.085Z"
      fill="white"
    />
    <path
      d="M197.958 288.94C195.397 288.94 192.836 287.962 190.883 286.007C186.98 282.1 186.983 275.769 190.891 271.865L233.799 229H207.958C203.871 229 200.195 226.513 198.675 222.719C197.155 218.925 198.097 214.587 201.054 211.766L264.054 151.645C268.05 147.832 274.379 147.979 278.192 151.975C282.005 155.971 281.857 162.3 277.862 166.113L232.921 209H257.958C262.003 209 265.65 211.438 267.198 215.176C268.745 218.913 267.888 223.216 265.025 226.074L205.025 286.014C203.073 287.964 200.516 288.94 197.958 288.94Z"
      fill="white"
    />
    <path
      d="M146.958 336.939C144.397 336.939 141.836 335.962 139.883 334.007C135.98 330.1 135.983 323.768 139.891 319.865L164.933 294.849C168.84 290.945 175.171 290.948 179.075 294.855C182.978 298.763 182.975 305.095 179.067 308.997L154.025 334.014C152.073 335.964 149.515 336.939 146.958 336.939Z"
      fill="white"
    />
    <path
      d="M55.958 303.939C53.397 303.939 50.8364 302.962 48.8833 301.007C44.98 297.1 44.9834 290.769 48.8906 286.865L64.6826 271.089C68.5898 267.187 74.9214 267.189 78.8247 271.096C82.728 275.003 82.7246 281.334 78.8174 285.237L63.0254 301.014C61.0732 302.964 58.5156 303.939 55.958 303.939Z"
      fill="white"
    />
    <path
      d="M234.958 317.939C232.397 317.939 229.836 316.962 227.883 315.007C223.98 311.1 223.983 304.769 227.891 300.865L279.883 248.926C283.79 245.022 290.122 245.025 294.025 248.933C297.928 252.84 297.925 259.171 294.018 263.074L242.025 315.014C240.073 316.964 237.515 317.939 234.958 317.939Z"
      fill="white"
    />
  </svg>
</div>
`;

exports[`WeatherIcon should return null 1`] = `<div />`;
